components/dashboard/src/App.tsx

```diff
@@ -177,13 +47,20 @@ function App() {
         };
     }, []);
 
+    // TODO: pull this redirect out of main react render if we only need to do it on page load
     // redirect to website for any website slugs
     if (isGitpodIo() && isWebsiteSlug(window.location.pathname)) {
         window.location.host = "www.gitpod.io";
         return <div></div>;
     }
 
-    if (isGitpodIo() && window.location.pathname === "/" && window.location.hash === "" && !loading && !user) {
+    if (loading) {
+        return <AppLoading />;
+    }
+
+    // TODO: Add some context to what this logic is for, does it have to happen every render, or just page load?
```

Comment:
only needs to be checked once on initial load

Reply:
I'm curious if this logic could be simpler and happen before we try to load any user. For example, if we just checked, for gitpod.io, if it's the root url w/ no hash, and there's no gitpod cookie we redirect to the www site.

```ts
// Redirect to www website if it's the root url and no cookie
if (window.location.pathname === "/" && window.location.hash === "") {
    // If there's no gp cookie, bounce to www site
    if (!GitpodCookie.isPresent(document.cookie)) {
        window.location.href = `https://www.gitpod.io`;
        return;
    }
}
```

Asking, cause I think if so, we could pull this out of the app component, and have a section in our top level `index.tsx` file that handles a few things like this before we try to render the app. It would simplify some of the logic in our app if so as well I think.